### PR TITLE
yield `addCardInEditMode` from {{content-kit-editor}}

### LINKED ISSUE #15: change utils/createComponentCard to create a card with display/edit modes
Currently the code to [swap between display and edit modes](https://github.com/bustlelabs/ember-content-kit/blob/06be5598cec2ef85c23ac4219fe07c3384aa25a8/addon/components/content-kit-component-card/template.hbs#L16-L20) of a card is managed by the [`isEditing` property](https://github.com/bustlelabs/ember-content-kit/blob/06be5598cec2ef85c23ac4219fe07c3384aa25a8/addon/components/content-kit-component-card/component.js#L45) of `content-kit-component-card` and `utils/createComponentCard` only creates cards with a [`display` key](https://github.com/bustlelabs/ember-content-kit/blob/06be5598cec2ef85c23ac4219fe07c3384aa25a8/addon/utils/create-component-card.js#L6-L19).

This should change to have display and edit properties that add/remove '-editor' from the card name and manage which component is shown directly. `content-kit-component-card` should probably be removed entirely.

### PR DESCRIPTION
**Pull request by bantic**
This allows adding a card that is rendered initially rendered in 'edit' mode instead of 'display' mode.

fixes #15
depends on this change to content-kit-editor: https://github.com/bustlelabs/content-kit-editor/pull/173 (merged as of content-kit-editor v 0.4.7)

Breaking changes:

  * remove `content-kit-component-card`

Changes `utils/createComponentCard` to create a card with display and edit modes set up instead of only a display mode. Previously, the now-removed `{{content-kit-component-card}}` template would use that component's own `isEditing` property to swap between rendering `{{card-name}}` and `{{card-name-editor}}` components.

Adds tests to ensure that the data is passed to cards appropriately
and that the object reference is broken.

### CODE REVIEW

File: tests/integration/components/content-kit-editor/component-test.js [hunk 1]
```diff
@@ -215,17 +217,134 @@ test('it de-links selected text and fires `on-change`', function(assert) {
...
+    {{/content-kit-editor}}
+  `);
+
+  this.$('button#add-card').click();
+
+  assert.ok(!this.$(`#demo-card`).length, 'Card added in edit mode');
+  assert.ok(this.$(`#demo-card-editor`).length, 'Card not in display mode');
+});
+
+test('#activeSectionTagNames is correct', function(assert) {
```
Comment (bantic) [inline tests/integration/components/content-kit-editor/component-test.js]: the `activeSectionTagNames` tests are not new; they were just in the wrong file before.

File: README.md [hunk 2]
```diff
@@ -221,18 +222,18 @@ will be used:
...
-* `saveAction`, an action accepting new data for the card payload, then saving
-  that data and toggling this card into display mode
-* `cancelAction`, an action toggling this card to display mode without saving
-  any changes.
+The component will be provided with the following `attrs`:
+
+  * `data`, the data payload for this card
+  * `editCard`, an action for toggling this card into edit mode (this action is a no-op if the card is already in edit mode)
+  * `removeCard`, an action for removing this card
+  * `saveCard`, an action accepting new data for the card payload, then saving
```
Comment (mixonic) [inline README.md]: Should these be split out as before, or at least be notated that `saveCard` for instance is not always present?

**Comment by mixonic** [comment]
Seems awesome besides the docs tweak!!

File: tests/integration/components/content-kit-editor/component-test.js [hunk 3]
```diff
@@ -237,17 +240,190 @@ test('it de-links selected text and fires `on-change`', function(assert) {
...
+            'has mutate-payload button');
+  assert.ok(this.$(`#demo-card:contains(${payload.foo})`).length,
+            'displays passed `data`');
+  this.$('button#mutate-payload').click();
+
+  assert.equal(passedPayload.foo, 'baz', 'mutates its payload');
+  assert.equal(payload.foo, 'bar', 'payload remains unchanged');
+});
+
+test('#activeSectionTagNames is correct', function(assert) {
```
Comment (bantic) [inline tests/integration/components/content-kit-editor/component-test.js]: the `activeSectionTagNames` tests are not new; they were in the wrong file before

File: README.md (hunk 2 above)
Comment (bantic) [inline README.md]: Currently all actions *are* passed to the card, but that is somewhat confusing. Added an issue to track this: https://github.com/bustlelabs/ember-content-kit/issues/18

**Pull request merged**